src/olmo_core/optim/config.py

```diff
@@ -97,42 +130,21 @@ def build_groups(
             else:
                 frozen_params.add(n)
 
-        if self.group_overrides is None:
-            return all_params.values()
-
-        # Build groups.
-        param_groups: List[Dict[str, Any]] = []
-        for g_idx, go in enumerate(self.group_overrides):
-            group: Dict[str, Any] = {"params": [], **go.opts}
-            for pattern in go.params:
-                matches = 0
-                for name in list(all_params.keys()):
-                    if fnmatch(name, pattern):
-                        group["params"].append(all_params.pop(name))
-                        matches += 1
-
-                if matches == 0:
-                    for name in frozen_params:
-                        if fnmatch(name, pattern):
-                            log.warning(
-                                f"optim group {g_idx} override pattern '{pattern}' matches a frozen parameter and will be ignored"
-                            )
-                            break
-                    else:
-                        msg = f"optim group {g_idx} override pattern '{pattern}' does not match any parameters"
-                        if strict:
-                            raise OLMoConfigurationError(msg)
-                        else:
-                            log.warning(msg)
-
-            if len(group["params"]) > 0:
-                param_groups.append(group)
-
-        # Put any left-over params into a default group.
-        if all_params:
-            param_groups.append({"params": list(all_params.values())})
-
-        return param_groups
+        group_overrides = [
+            self._expand_param_globs(go, all_params, frozen_params, g_idx, strict=strict)
+            for g_idx, go in enumerate(self.group_overrides or [])
+        ]
+
+        # Treat no overrides as its own override group
+        overriden_param_names = {name for go in group_overrides for name in go.params}
+        default_override = OptimGroupOverride(list(all_params.keys() - overriden_param_names), {})
```

Comment:
The order of parameters within a group should be deterministic to be safe, and I think this line breaks that. Could change to something like:

```suggestion
        default_override = OptimGroupOverride([name for name in all_params.keys() if name not in overriden_param_names], {})
```